tests/test_format.py

```diff
@@ -0,0 +1,178 @@
+import xarray as xr
+
+import xarray_regrid
+from xarray_regrid.utils import format_for_regrid
+
+
+def test_covered():
+    dx_source = 2
+    source = xarray_regrid.Grid(
+        north=90,
+        east=360,
+        south=-90,
+        west=0,
+        resolution_lat=dx_source,
+        resolution_lon=dx_source,
+    ).create_regridding_dataset()
+
+    dx_target = 1
+    target = xarray_regrid.Grid(
+        north=80,
+        east=350,
+        south=-80,
+        west=10,
+        resolution_lat=dx_target,
+        resolution_lon=dx_target,
+    ).create_regridding_dataset()
+
+    formatted = format_for_regrid(source, target)
+
+    # Formatting utils shouldn't modify this one at all
+    xr.testing.assert_equal(source, formatted)
+
+
+def test_no_edges():
+    dx_source = 2
+    source = xarray_regrid.Grid(
+        north=90 - dx_source / 2,
+        east=360 - dx_source / 2,
+        south=-90 + dx_source / 2,
+        west=0 + dx_source / 2,
+        resolution_lat=dx_source,
+        resolution_lon=dx_source,
+    ).create_regridding_dataset()
+
+    dx_target = 1
+    target = xarray_regrid.Grid(
+        north=90,
+        east=360,
+        south=-90,
+        west=0,
+        resolution_lat=dx_target,
+        resolution_lon=dx_target,
+    ).create_regridding_dataset()
+
+    formatted = format_for_regrid(source, target)
+
+    # Should add wraparound and polar padding rows/columns
+    assert formatted.latitude[0] == -90
+    assert formatted.latitude[-1] == 90
+    assert formatted.longitude[0] == -1
+    assert formatted.longitude[-1] == 361
+    assert (formatted.longitude.diff("longitude") == 2).all()
+
+
+def test_360_to_180():
```

Comment:
Would it be useful here to parameterize the test by `dx_source` (and maybe dx_target)?

Also, these tests only test for regularly spaced grids, what if a grid has irregular spacing?

Reply:
These would be good additions, I can follow up with another PR to cover more test cases.